### PR TITLE
JCRVLT-526 improved support of legacy timezone designators "+hhmm"

### PR DESCRIPTION
**Pull request by kwin**
Based on a suggestion by Julian Sedding

### CODE REVIEW

File: vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/impl/PackagePropertiesImpl.java [hunk 1]
```diff
@@ -224,21 +222,17 @@ public boolean requiresRestart() {
...
-                        String reformattedDate = splittedDate.group("dateUptoTzHours") + ":" + tzminutes;
-                        result = ISO8601.parse(reformattedDate);
-                    }
+                ZonedDateTime zonedDateTime;
+                try {
+                    // supports date in format new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX")
+                    zonedDateTime = ZonedDateTime.parse(p, DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+                } catch (DateTimeParseException e) {
+                    // support dates given out via new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ"); as well (used in package-maven-plugin till version 1.0.3, compare with https://issues.apache.org/jira/browse/JCRVLT-276)
+                    zonedDateTime = ZonedDateTime.parse(p, DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSX"));
```
Comment (stoerr) [inline vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/impl/PackagePropertiesImpl.java:233]: That undoubtedly looks much cleaner than the pattern replacement approach. So, if that is backwards compatible to ISO8601 and does parse the +0200 timezone format variant, then that's completely fine with me. Mind you - the test I wrote did not intend to see whether it still parses all date formatting variants ISO8601.parse understands, if there are some, since it was still using ISO8601.parse so far. So you might want to check whether you need to extend the test somewhat.

File: vault-core/src/test/java/org/apache/jackrabbit/vault/packaging/impl/PackagePropertiesImplTest.java [hunk 2]
```diff
@@ -93,9 +94,6 @@ public void testDateFormat() {
         checkDateParsing("2021-05-26T15:12:21.673+0200","2021-05-26T13:12:21.673Z");
         checkDateParsing("2021-05-26T15:12:21.673-0230","2021-05-26T17:42:21.673Z");
 
-        // missing timezone is treated as UTC. TODO is that right?
-        checkDateParsing("2021-05-26T15:12:21.673","2021-05-26T15:12:21.673Z");
-
```
Comment (stoerr) [inline vault-core/src/test/java/org/apache/jackrabbit/vault/packaging/impl/PackagePropertiesImplTest.java:98]: What happens now if the timezone is missing? I don't know whether anything produces that date format, but at least I'm a fan of specifying such edge cases in the test, even if the documentation doesn't really say anything about that. Not sure whether it's a problem if that behavior changes. That throws an exception now, right?

**Review by jsedding — APPROVED** [review]
@kwin Thanks! This looks a lot better than parsing/fixing patterns with regexp.

Should we create a constant for `DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSX")`? I don't know how costly it is to recreate, and afaik DTFs are thread-safe.

I wonder if a `DateTimeFormatter` could be created to correctly parse all the expected inputs, but I couldn't quickly find a configuration that passes the tests.

File: vault-core/src/test/java/org/apache/jackrabbit/vault/packaging/impl/PackagePropertiesImplTest.java (hunk 2 above)
Comment (kwin) [reply vault-core/src/test/java/org/apache/jackrabbit/vault/packaging/impl/PackagePropertiesImplTest.java:98]: That date format was never produced and therefore is not supported (i.e. returns null)

Comment (stoerr) [reply vault-core/src/test/java/org/apache/jackrabbit/vault/packaging/impl/PackagePropertiesImplTest.java:98]: It didn't return null, as the test says. But never mind.
